node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/QuestMetricsDb.java

```diff
@@ -236,122 +199,76 @@ public void close() {
             engine.close();
     }
 
-    /**
-     * Repairs this db on corruption.
-     *
-     * @param e the exception indicating corruption
-     */
-    private void repair(Exception e) {
-        log.log(Level.WARNING, "QuestDb seems corrupted, wiping data and starting over", e);
-        IOUtils.recursiveDeleteDir(new File(dataDir));
-        initializeDb();
-    }
-
-    private boolean exists(String table, SqlExecutionContext context) {
-        return 0 == engine.getStatus(context.getCairoSecurityContext(), new Path(), table);
-    }
-
     private void ensureTablesExist() {
-        SqlExecutionContext context = newContext();
-        if (exists(nodeTable, context))
-            ensureNodeTableIsUpdated(context);
+        if (nodeTable.exists())
+            ensureNodeTableIsUpdated();
         else
-            createNodeTable(context);
+            createNodeTable();
 
-        if (exists(clusterTable, context))
-            ensureClusterTableIsUpdated(context);
+        if (clusterTable.exists())
+            ensureClusterTableIsUpdated();
         else
-            createClusterTable(context);
+            createClusterTable();
+    }
+
+    private void ensureNodeTableIsUpdated() {
+        try {
+            if (0 == engine.getStatus(newContext().getCairoSecurityContext(), new Path(), nodeTable.name)) {
```

Comment:
This is just `nodeTable.exists()`, but that is already checked before calling this the method?

Reply:
Yes, it does nothing now, but I need it each time a column is added.

Reply:
Yeah, I get that, it's just that this `if` is redundant, could leave the `try` block empty until you need it

Reply:
Good point, done